src/iotEdgeExplorer.ts

```diff
@@ -169,7 +199,9 @@ export class IoTEdgeExplorer extends BaseExplorer {
             }
 
             try {
-                const isValid = await this.isValidDeploymentJsonSchema(contentJson);
+                const isValid = await this.isValidDeploymentJsonSchema(contentJson)
+                    && this.isValidCreateOptions(contentJson.modulesContent.$edgeAgent["properties.desired"]);

```

Comment:
properties.desired has lots of entries right? Pass the how desired property as modules?